kratos/utilities/divide_tetrahedra_3d_4.cpp

```diff
@@ -229,5 +230,94 @@ void DivideTetrahedra3D4::GenerateIntersectionsSkin() {
             KRATOS_ERROR << "Trying to generate the intersection skin in DivideTetrahedra3D4::GenerateIntersectionsSkin() for a non-split element.";
         }
     };
+
+    void DivideTetrahedra3D4::GenerateExteriorFaces(
+        std::vector < IndexedPointGeometryPointerType > &rExteriorFacesVector,
+        std::vector < unsigned int > &rExteriorFacesParentSubdivisionsIdsVector,
+        const std::vector < IndexedPointGeometryPointerType > &rSubdivisionsContainer) {
+
+        // Set some geometry constant parameters
+        const unsigned int n_faces = 4;
+
+        // Set the exterior faces vectors
+        rExteriorFacesVector.clear();
+        rExteriorFacesParentSubdivisionsIdsVector.clear();
+
+        // Iterate the triangle faces
+        for (unsigned int i_face = 0; i_face < n_faces; ++i_face) {
+            std::vector < unsigned int > aux_ext_faces_parent_ids;
+            std::vector < DivideTetrahedra3D4::IndexedPointGeometryPointerType > aux_ext_faces;
+
+            DivideTetrahedra3D4::GenerateExteriorFaces(
+                aux_ext_faces,
+                aux_ext_faces_parent_ids,
+                rSubdivisionsContainer, 
+                i_face);
+            
+            rExteriorFacesVector.insert(rExteriorFacesVector.end(), aux_ext_faces.begin(), aux_ext_faces.end());
+            rExteriorFacesParentSubdivisionsIdsVector.insert(rExteriorFacesParentSubdivisionsIdsVector.end(), aux_ext_faces_parent_ids.begin(), aux_ext_faces_parent_ids.end());
+        }
+    };
+
+    void DivideTetrahedra3D4::GenerateExteriorFaces(
+        std::vector < IndexedPointGeometryPointerType > &rExteriorFacesVector,
+        std::vector < unsigned int > &rExteriorFacesParentSubdivisionsIdsVector,
+        const std::vector < IndexedPointGeometryPointerType > &rSubdivisionsContainer,
+        const unsigned int FatherFaceId) {
+        // Set some geometry constant parameters
+        const unsigned int n_faces = 4;
+
+        // Set the exterior faces vector
+        rExteriorFacesVector.clear();
+        rExteriorFacesParentSubdivisionsIdsVector.clear();
+
+        if (mIsSplit) {
+            // Create the face nodes data
+            // The position represents the face while the value the real and intersection nodes in that face edges
+            std::vector < std::vector < unsigned int > > edges_map = {
```

Comment:
maybe you could use a std::array here

Reply:
i mean 

         std::array< std::array< unsinged int> >

Reply:
Done. The unique thing to comment is that double brackets are needed to avoid a compilation warning (I found that this is supposed to be fixed in c++14).